### PR TITLE
Updates to the firewall pages

### PR DESCRIPTION
**Pull request by Nephiaust**
Started updating the firewall pages, to explain how they work.

Updated the general, legacy and zone pages.

Need to do more work on the explainations, but wanted a review of the current work to see if its acceptable.

### CODE REVIEW

**Comment by Nephiaust** [comment]
@rebortg if you are happy with the style, I will continue updating the documentation. I will also fix the linting issues as well as part of the documentation updates.

File: docs/configuration/firewall/general-legacy.rst [hunk 1]
```diff
@@ -1,10 +1,10 @@
 :lastproofread: 2021-06-29
 
-.. _firewall-legacy:
```
Comment (rebortg) [inline docs/configuration/firewall/general-legacy.rst:3]: if you change the reference name, you must check the usage. For example it is use [here](https://github.com/vyos/vyos-documentation/blob/a7f6b67f7e8c2d6aae9db17832ed26abbb29beee/docs/quick-start.rst?plain=1#L134)

File: docs/configuration/firewall/general.rst [hunk 2]
```diff
@@ -1,10 +1,10 @@
-:lastproofread: 2021-06-29
+:lastproofread: 2023-09-17
 
-.. _firewall:
+.. _firewall-configuration:
```
Comment (rebortg) [inline docs/configuration/firewall/general.rst:3]: here the same behavior with the reference name like in the legacy version

**Review by rebortg — CHANGES_REQUESTED** [review]

File: docs/configuration/firewall/general-legacy.rst (hunk 1 above)
Comment (Nephiaust) [reply docs/configuration/firewall/general-legacy.rst:3]: Excluding those issues, happy with the overall style and layout? If so I will start working on fixing it all up properly (including the references).

**Comment by nicolas-fort** [comment]
At first sight, content looks! It's good and necessary to improve vyos documentation. Thanks in advance for this work!

**Comment by rebortg** [comment]
i'm happy with all contribution. :-) 
It is all written well. All commands should be written in `.. cfgcmd:: ` style, but its totally fine for the moment. I have a rewrite of the documentation style guide on my todo list. And will also add some vyos icons for drawings.

**Pull request merged**